modules/adopter-registration-impl/src/main/java/org/opencastproject/adopter/statistic/ScheduledDataCollector.java

```diff
@@ -415,8 +416,15 @@ public void setOrganizationDirectoryService(OrganizationDirectoryService orgDirS
     this.organizationDirectoryService = orgDirServ;
   }
 
-  @Reference(cardinality = ReferenceCardinality.OPTIONAL)
+  @Reference(
+      cardinality = ReferenceCardinality.OPTIONAL,
+      policy = ReferencePolicy.DYNAMIC,
+      unbind = "unsetTobiraEndpoint")
   public void setTobiraEndpoint(TobiraEndpoint endpoint) {
     this.tobiraEndpoint = endpoint;
   }
+
+  public void unsetTobiraEndpoint(TobiraEndpoint endpoint) {
```

Comment:
Is this argument here actually useful? :thinking: Is it always `null`?

Reply:
It's not actually null, it's the service being unregistered :)

Reply:
But like, what's the value of `endpoint` when the function is called?

Reply:
`endpoint` is the service instance which is shutting down - `TobiraEndpoint` in this case.  This is a simplification of the `1..n` relationship where *multiple* `TobiraEndpoint`s could be defined.  We'd need to pass the one which is going away so that the binding service (`ScheduledDataCollector`) can do whatever it needs to shut down its end of the connection.

Reply:
Thanks for the explanation! This can be resolved then.